module-apps/apps-common/widgets/TimeFixedWidget.cpp

```diff
@@ -36,125 +37,152 @@ namespace gui
-            firstPrimoText->setSize(widestDigit, hBox->widgetArea.h);
-            firstPrimoText->setText(std::to_string(first / 10));
-            firstSecundoText->setText(std::to_string(first % 10));
+        leftBox.box = new gui::HBox(mainBox, 0, 0, 0, 0);
+        leftBox.box->setAlignment(gui::Alignment(gui::Alignment::Horizontal::Right, gui::Alignment::Vertical::Center));
+        leftBox.box->setEdges(gui::RectangleEdge::None);
+
+        attachLabelToBox(colon, mainBox);
+        setColon();
+
+        rightBox.box = new gui::HBox(mainBox, 0, 0, 0, 0);
+        rightBox.box->setAlignment(gui::Alignment(gui::Alignment::Horizontal::Left, gui::Alignment::Vertical::Center));
+        rightBox.box->setEdges(gui::RectangleEdge::None);
+
+        attachLabelToBox(leftBox.minus, leftBox.box);
+        setMinus();
+
+        for (auto &digit : leftBox.digits) {
+            attachLabelToBox(digit, leftBox.box);
         }
-        else {
-            firstPrimoText->setText("");
-            firstPrimoText->setSize(0, 0);
-            firstSecundoText->setText(std::to_string(first));
+
+        for (auto &digit : rightBox.digits) {
+            attachLabelToBox(digit, rightBox.box);
         }
-        firstHBox->resizeItems();
+
+        mainBox->resizeItems();
     }
 
-    void TimeFixedWidget::setSecond(const uint32_t second)
+    void TimeFixedWidget::attachLabelToBox(Label *&label, HBox *&box) const
     {
-        const auto decimal = second / 10;
-        if (decimal != 0u) {
-            secondPrimoText->setText(std::to_string(second / 10));
-            secondSecundoText->setText(std::to_string(second % 10));
-        }
-        else {
-            secondPrimoText->setText("0");
-            secondSecundoText->setText(std::to_string(second));
+        label = new gui::Label(box, 0, 0, 0, 0);
+        label->setEdges(gui::RectangleEdge::None);
+        label->setAlignment(gui::Alignment(gui::Alignment::Horizontal::Center, gui::Alignment::Vertical::Center));
+    }
+
+    void TimeFixedWidget::setMinus() const
+    {
+        leftBox.minus->setText(minusVisible ? minusSign : "");
+    }
+
+    void TimeFixedWidget::setColon() const
+    {
+        colon->setText(colonSign);
+        colon->activeItem = false;
+    }
+
+    void TimeFixedWidget::setLeftBox(uint32_t minutes) const
+    {
+        bool isLimitExceed = minutes > 999;
+        minutes            = isLimitExceed ? minutes % 1000 : minutes;
+
+        setDigits(std::to_string(minutes), leftBox.digits);
+
+        leftBox.box->resizeItems();
+    }
+
+    void TimeFixedWidget::setRightBox(uint32_t seconds) const
+    {
+        bool isLimitExceed = seconds > 99;
+        seconds            = isLimitExceed ? seconds % 100 : seconds;
+
+        bool shouldBeDisplayedAsOneDigit = seconds < 10;
+        std::string text = shouldBeDisplayedAsOneDigit ? "0" + std::to_string(seconds) : std::to_string(seconds);
+
+        setDigits(std::move(text), rightBox.digits);
```

Comment:
```suggestion
stringstream ss;
ss << setfill('0') << setw(2) << seconds;

        setDigits(ss.str(), rightBox.digits);
```

Reply:
similar situation to:
     -   bool isLimitExceed = minutes > 999;
     +  minutes            = isLimitExceed ? minutes % 1000 : minutes;
     +  minutes            = minutes % 1000;
  
Less expresive but suggested implementatnon is more compact